### ISSUE TITLE
No option to skip wrapping closures for select classes

### ISSUE THREAD

**Issue by davidbyoung** [opis/closure#51]
I'm raising a new issue for https://github.com/opis/closure/issues/45#issuecomment-576454221 as I believe it's a separate feature request altogether.  Some classes, like [my DI container](https://github.com/aphiria/aphiria/blob/master/src/DependencyInjection/src/Container.php#L49), have logic that essentially prevents them from being serialized.  However, if I am serializing a closure that uses the container, eg `fn () => $container->resolve(Foo::class)`, Opis will attempt to recursively wrap all scoped variables' closures via reflection.  It would be nice to have some way of preventing Opis from wrapping specific class' closure properties.

**Comment by davidbyoung** [comment]
@msarca Does this seem like a doable feature?  Unfortunately, I'm blocked on further development, and am wondering if I need to try rolling my own solution.  I tried playing around with a PR to add this sort of functionality, but I am not confident that I handled edge cases properly.  Thanks!

**Comment by msarca** [comment]
It is doable, but I'm not yet sure what would be the best way to implement this. We also have our own DI container solution and preventing wrapping would be something desirable for us too. 

One approach on this would be to define 2 new static methods:

```php
// Skip closure wrapping for Foo, 
// and continue it on its parent class, if it has one
SerializableClosure::skipClosureWrapping(Foo::class);

// Stop closure wrapping at Bar
SerializableClosure::stopClosureWrapping(Bar::class);
```

The above approach is perfect when we're dealing with 3rd party code, over which we lack control. But it would be terrible to ask our users to write some bootstrap code if they want to use our container library properly. That's why a secondary mechanism must exist.

```php
use Opis\Closure\Traits\{SkipClosureWrapping, StopClosureWrapping};

class Foo
{
     use SkipClosureWrapping;
}

class Bar 
{
    use StopClosureWrapping;
}
```

Any thoughts?

**Comment by davidbyoung** [comment]
It would be nice to have both options.  For example, my DI container library isn't even aware of opis/closure, but other libraries that compose my container might be.  So, allowing me to opt out of wrapping closures from outside the DI container would be great.

**Comment by davidbyoung** [comment]
Do you know if there's any plan to support this soon?  I am still blocked by this.  Thanks!

**Comment by msarca** [comment]
The wrapping mechanism will no longer exist in the upcoming major release that targets PHP >7.4 #59
